test/unit/test_jinja.py

```diff
@@ -1,11 +1,378 @@
+        returns("'1'"),
+        returns("'1'"),
+    ),
+    (
+        '''foo: "'{{ '1' }}'"''',
+        returns("'1'"),
+        returns("'1'"),
+    ),
+    (
+        '''foo: "{{ 1 | as_text }}"''',
+        returns('1'),
+        returns('1'),
+    ),
+    (
+        '''foo: "{{ 1 | as_bool }}"''',
+        returns('1'),
+        raises(JinjaRenderingException),
+    ),
+    (
+        '''foo: "{{ 1 | as_number }}"''',
+        returns('1'),
+        returns(1),
+    ),
+    (
+        '''foo: "{{ 1 | as_native }}"''',
+        returns('1'),
+        returns(1),
+    ),
+    (
+        '''foo: "{{ '1' | as_text }}"''',
+        returns('1'),
+        returns('1'),
+    ),
+    (
+        '''foo: "{{ '1' | as_bool }}"''',
+        returns('1'),
+        raises(JinjaRenderingException),
+    ),
+    (
+        '''foo: "{{ '1' | as_number }}"''',
+        returns('1'),
+        returns(1),
+    ),
+    (
+        '''foo: "{{ '1' | as_native }}"''',
+        returns('1'),
+        returns(1),
+    ),
+    # booleans.
+    # Note the discrepancy with true vs True: `true` is recognized by jinja but
+    # not literal_eval, but `True` is recognized by ast.literal_eval.
+    # For extra fun, yaml recognizes both.
+    # unquoted true
+    (
+        '''foo: "{{ True }}"''',
+        returns('True'),
+        returns('True'),
+    ),
+    (
+        '''foo: "{{ True | as_text }}"''',
+        returns('True'),
+        returns('True'),
+    ),
+    (
+        '''foo: "{{ True | as_bool }}"''',
+        returns('True'),
+        returns(True),
+    ),
+    (
+        '''foo: "{{ True | as_number }}"''',
+        returns('True'),
+        raises(JinjaRenderingException),
+    ),
+    (
+        '''foo: "{{ True | as_native }}"''',
+        returns('True'),
+        returns(True),
+    ),
+    # unquoted true
+    (
+        '''foo: "{{ true }}"''',
+        returns("True"),
+        returns("True"),
+    ),
+    (
+        '''foo: "{{ true | as_text }}"''',
+        returns("True"),
+        returns("True"),
+    ),
+    (
+        '''foo: "{{ true | as_bool }}"''',
+        returns("True"),
+        returns(True),
+    ),
+    (
+        '''foo: "{{ true | as_number }}"''',
+        returns("True"),
+        raises(JinjaRenderingException),
+    ),
+    (
+        '''foo: "{{ true | as_native }}"''',
+        returns("True"),
+        returns(True),
+    ),
+    (
+        '''foo: "{{ 'true' | as_text }}"''',
+        returns("true"),
+        returns("true"),
+    ),
+    # quoted 'true'
+    (
+        '''foo: "'{{ true }}'"''',
+        returns("'True'"),
+        returns("'True'"),
+    ),  # jinja true -> python True -> str(True) -> "True" -> quoted
+    (
+        '''foo: "'{{ true | as_text }}'"''',
+        returns("'True'"),
+        returns("'True'"),
+    ),
+    (
+        '''foo: "'{{ true | as_bool }}'"''',
+        returns("'True'"),
+        returns("'True'"),
+    ),
+    (
+        '''foo: "'{{ true | as_number }}'"''',
```

Comment:
This one is surprising to me - I'd think that `as_number` would fail because `true` is not numeric. I imagine I'm missing some quirk here - just curious what it might be!

Reply:
It's because of the quotes. rendering goes down the `"".join(...)` path because there are multiple "nodes" in the raw render result.

We could definitely make `NumberMarker`/`BoolMarker` examine their inputs and error out here if we want. It would potentially be something silly like: try to literal_eval, make sure the result is a number/bool and error out of not, then throw away that result (or store it and expose it to the concatenation function later). It seemed like a lot of complexity of dubious value since things will end up as strings anyway, but I can see why it'd be desirable.

That's all native mode. In text mode, the `as_*` are all no-ops so they do nothing.

Reply:
ok - that's kind of what i figured. The filter in this context is pretty useless anyway given the wrapping quotes outside of the jinja expression. Works for me!